Misc/NEWS.d/next/Library/2024-07-17-17-51-25.gh-issue-121331.8Gk6LP.rst

```diff
@@ -0,0 +1 @@
+:func:`inspect.getsource` now works for genexp code objects, no longer only returning the first line of a multi-line expression, and now only returning the code of the generator expression itself, not the whole line of code it is created on, allowing for disambiguation when multiple generator expressions are defined on the same line.
```

Comment:
Tiny nitpick: "genexp" sounds to me like a more technical term that others might not understand. It's probably a better idea to just say "generator expression."
```suggestion
:func:`inspect.getsource` now works for generator expressions, no longer only returning the first line of a multi-line expression, and now only returning the code of the generator expression itself, not the whole line of code it is created on, allowing for disambiguation when multiple generator expressions are defined on the same line.
```

Although, this is quite a long news entry for a bugfix. It could be as simple as "`inspect.getsource` now works properly with generator expressions." -- I'll let you decide, though.

Reply:
Good point, though I think "generator expression code objects" is probably a more precise wording as `getsource` does not (and not does it puport to) work on generator expression objects themselves, only the code object returned from calling `.gi_code` on the generator expression. Unless any other objections, I will change it to that and remove the following sentences.